src/main/java/uk/ac/ebi/ega/accession/file/rest/HashValidator.java

```diff
@@ -34,16 +34,17 @@ public boolean isValid(FileModel fileModel, ConstraintValidatorContext constrain
 
         constraintValidatorContext.disableDefaultConstraintViolation();
 
-        if (fileModel.getHashType() == null) {
-            constraintValidatorContext.buildConstraintViolationWithTemplate(
-                    "Please provide a hashType.Supported types :MD5,SHA1").addConstraintViolation();
+        if (fileModel.getFileMd5() == null || fileModel.getFileMd5().length() != 32 ||
+            !fileModel.getFileMd5().equals(fileModel.getFileMd5().toLowerCase())) {
+            constraintValidatorContext.buildConstraintViolationWithTemplate("Please provide a valid lowercase hash for md5")
+                    .addConstraintViolation();
             return false;
         }
 
-        if (fileModel.getHash() == null || (fileModel.getHashType() == HashType.MD5 && fileModel.getHash().length() != 32) ||
-                (fileModel.getHashType() == HashType.SHA1 && fileModel.getHash().length() != 40)) {
-            constraintValidatorContext.buildConstraintViolationWithTemplate("Please provide a valid hash")
-                    .addConstraintViolation();
+        if (fileModel.getFileSha2() != null && (fileModel.getFileSha2().length() != 0 && fileModel.getFileSha2().length() != 64 ) ||
```

Comment:
This condition will NPE when sha2 is null. It is an or, so both conditions need to be checked.